post-processing/post_processing.py

```diff
@@ -43,33 +49,107 @@ def run_post_processing(self, path, figures_of_merit):
+        # check for required columns
+        if (len(columns) < 3) | (False in required_column_matches):
+            raise KeyError("Config must contain at least 3 specified columns: a figure of merit value, a figure of merit unit, and something to plot the figure of merit against", columns)
+
+        invalid_columns = []
+        # check for invalid columns
+        for col in columns:
+            if col not in df.columns:
+                invalid_columns.append(col)
+        if invalid_columns:
+            raise KeyError("Could not find columns", invalid_columns)
+
+        filters = config["filters"]
+        mask = pd.Series(df.index.notnull())
+        # filter rows
+        if filters:
+            mask = reduce(op.and_, (self.row_filter(f, df) for f in filters))
+        if df[mask].empty:
+            raise pd.errors.EmptyDataError("Filtered dataframe is empty", df[mask].index)
+
+        print("")
         print("Selected dataframe:")
-        print(df[columns])
+        print(df[columns][mask])
+        print("")
 
         # call a plotting script
-        # TODO
+        # TODO: plot(df, config)
 
         if self.debug & self.verbose:
             print("Full dataframe:")
             print(df.to_json(orient="columns", indent=2))
 
-        return df[columns]
+        return df[columns][mask]
+
+    # operator lookup dictionary
+    op_lookup = {
+        "==":   op.eq,
+        "!=":   op.ne,
+        "<" :   op.lt,
+        ">" :   op.gt,
+        "<=":   op.le,
+        ">=":   op.ge
+    }
+
+    def row_filter(self, filter, df: pd.DataFrame):
+        """
+            Return a dataframe mask based on a filter condition. The filter is a list that contains a column name, an operator, and a value (e.g. ["flops_value", ">=", 1.0]).
+
+            Args:
+                filter: list, a condition based on which a dataframe is filtered.
+                df: dataframe, used to create a mask by having the filter condition applied to it.
+        """
+
+        column, str_op, value = filter
+        if self.debug:
+            print("Applying row filter condition:", column, str_op, value)
+
+        # check column validity
+        if column not in df.columns:
+            raise KeyError("Could not find column", column)
+
+        # check operator validity
+        operator = self.op_lookup.get(str_op)
+        if operator is None:
+            raise KeyError("Unknown comparison operator", str_op)
+
+        # evaluate expression and extract dataframe mask
+        if value is None:
+            if operator == op.eq:
+                mask = df[column].isnull()
+            else:
+                mask = df[column].notnull()
+        else:
+            try:
+                # dataframe column is interpreted as the same type as the supplied value
+                mask = operator(df[column].astype(type(value)), value)

```

Comment:
There's a slight issue here, if the type of values of the column doesn't agree with the type of the filter value from the YAML (eg. int vs float). But this is part of a larger issue, that will be resolved separately, see #176